docs/architecture/adr-022-per-subspace-token-factory.md

```diff
@@ -0,0 +1,228 @@
+# ADR 022: Per-subspace token factory
+
+## Changelog
+- April 28th, 2023: First draft;
+
+## Status
+
+Draft
+
+## Abstract
+
+This ADR introduces a new feature that enables the subspace admins to create, mint and burn a subspace related
+coin. 
+
+## Context
+
+Desmos is a social network protocol that allows users to create, share, and engage with content 
+on a decentralized platform. 
+Currently, to post or interact with the content in a subspace you need to pay the gas fee using DSM. 
+This have the consequence of not giving any financial incentive on creating a social network on Desmos.
+
+## Decision
+
+We will wrap the Osmosis [Token Factory module](https://docs.osmosis.zone/osmosis-core/modules/tokenfactory/)
+with the following modifications:
+1. Instead of use the token creator address to compose the coin denom we will use the subspace treasury address
+   as following: `factory/{trasury_address}/subdenom`;
+2. All the operations of `CreateDenom`, `Mint`, `Burn`, `SetDenomMetadata`, `SetBeforeSendHook` and
+   `ForceTransfer` can only be performed by subspace treasury;
+3. The `CreateDenom` action will burn the coins instead of send them to the community pool.  
```

Comment:
I'm actually very hesitant in deciding whether we should burn the paied fee, or make the token creation free. 

By **burning** the fees, we are adding yet-another DSM burning mechanism which, combined with the burning of the fees while setting a custom subspace fee token (ADR-21), will increment the burning rate of DSM. However, I think if requiring users all these DSM first to create the token and then to set it as a fee token is actually something good to do. 

By **not burning** we are making it easier for users to create their tokens, but we are opening up to possible spam attacks. 

What are your thoughts @dadamu @manu0466?

Reply:
To prevent spam attacks issue, we can leverage `x/fees` to have a proposal to define the fees of `CreateDenom`, then have burning mechanism on `x/fees` in the future. What do you think?

Reply:
I think that to prevent spam attacks it's better if the `CreateDenom` operation requires the subspace admin to **burn** some DSM. This will prevent spam attacks and disincetives people from creating denoms without a strong reason also sisnce we are issuing a new coin on Desmos it's better from a tokenomics prospective to burn DSM instead of pay a fee to issue a new denom.